src/main/java/backend/Logic.java

```diff
@@ -200,18 +198,77 @@ public Model getRepo(String repoId) {
      * @return true if label replacement on GitHub was a success, false otherwise.
      */
     public CompletableFuture<Boolean> replaceIssueLabels(TurboIssue issue, List<String> newLabels) {
+        List<String> originalLabels = issue.getLabels();
+
         logger.info("Changing labels for " + issue + " on UI");
-        issue.setLabels(newLabels);
+        /* Calls models to replace the issue's labels locally since the the reference to the issue here
+           could be invalidated by changes to the models elsewhere */
+        Optional<TurboIssue> localReplaceResult =
+                models.replaceIssueLabels(issue.getRepoId(), issue.getId(), newLabels);
+        if (!localReplaceResult.isPresent()) {
+            return CompletableFuture.completedFuture(false);
+        }
         refreshUI();
 
-        return updateIssueLabelsOnRepo(issue, newLabels);
+        return updateIssueLabelsOnServer(issue, newLabels)
+                .thenApply((isUpdateSuccessful) -> handleIssueLabelsUpdateOnServerResult(
+                            isUpdateSuccessful, localReplaceResult.get(), originalLabels));
+    }
+
+    /**
+     * Gets the issue identified by {@code repoId} and {@code issueId} in {@link Logic#models}
+     * @param repoId
+     * @param issueId
+     * @return
+     */
+    private Optional<TurboIssue> getIssue(String repoId, int issueId) {
```

Comment:
By right, it should be something seriously wrong but for our codes at the moment, there is a slim chance that it might happen. For example, during update reconciliation, an issue is removed from a `Model`'s issues list but before the replacement can be put in, the program is context switched to replacing labels which does an issue look up as above but won't find the issues.